ui/src/app/pages/data/data-set/export-dataset-modal.tsx

```diff
@@ -0,0 +1,252 @@
+          prePackagedConceptSet: dataset.prePackagedConceptSet
+        })
+      };
+    }
+
+    function loadHtmlStringIntoIFrame(html) {
+      const placeholder = document.createElement('html');
+      placeholder.innerHTML = html;
+      // Some styling changes to the jupyter notebook to make it easier to view
+      placeholder.style.overflowY = 'scroll';
+      placeholder.getElementsByTagName('body')[0].style.overflowY = 'scroll';
+      placeholder.querySelector<HTMLElement>('#notebook').style.paddingTop = '0';
+
+      // Remove input column from notebook cells. Also possible to strip this out in Calhoun but requires some API changes
+      placeholder.querySelectorAll('.input_prompt').forEach(e => e.remove());
+      return <iframe scrolling='no' style={{width: '100%', height: '100%', border: 'none'}} srcDoc={placeholder.outerHTML}/>;
+    }
+
+    function loadCodePreview(kernel: KernelTypeEnum) {
+      setIsLoadingNotebook(true);
+      setErrorMsg(null);
+      dataSetApi().previewExportToNotebook(workspace.namespace, workspace.id, {
+        dataSetRequest: createDataSetRequest(),
+        kernelType: kernel,
+        newNotebook: false,
+        notebookName: '',
+      }).then(resp => setCodePreview(loadHtmlStringIntoIFrame(resp.html)))
+        .catch(() => setErrorMsg('Could not load code preview. Please try again or continue exporting to a notebook.'))
+        .finally(() => setIsLoadingNotebook(false));
+    }
+
+    function onCodePreviewClick() {
+      if (codePreview) {
+        setCodePreview(null);
+      } else {
+        AnalyticsTracker.DatasetBuilder.SeeCodePreview();
+        loadCodePreview(kernelType);
+      }
+    }
+
+    function onNotebookSelect(v) {
+      setCreatingNewNotebook(v === '');
+      setNotebookName(v);
+      setErrorMsg(null);
+
+      if (v === '') {
+        setCreatingNewNotebook(true);
+      } else {
+        setCreatingNewNotebook(false);
+        setIsLoadingNotebook(true);
+        workspacesApi().getNotebookKernel(workspace.namespace, workspace.id, v)
+          .then(resp => setKernelType(resp.kernelType))
+          .catch(() => setErrorMsg('Could not fetch notebook metadata. Please try again or create a new notebook.'))
+          .finally(() => setIsLoadingNotebook(false));
+      }
+    }
+
+    useEffect(() => {
+      workspacesApi().getNoteBookList(workspace.namespace, workspace.id)
+        .then(notebooks => setExistingNotebooks(notebooks))
+        .catch(() => setExistingNotebooks([])); // If the request fails, at least let the user create new notebooks
+    }, [workspace]);
+
+    useEffect(() => {
+      if (codePreview) {
+        loadCodePreview(kernelType);
+      }
+    }, [kernelType]);
+
+    const errors = {
+      ...validate({notebookName}, {
+        notebookName: {
+          presence: {allowEmpty: false},
+          exclusion: {
+            within: creatingNewNotebook && existingNotebooks ? existingNotebooks.map(fd => fd.name.slice(0, -6)) : [],
```

Comment:
what does `fd` mean here?

Reply:
FileDetail. I refactored this so that the name is pulled out upon setting the existingNotebooks value.